tests/unit/test_util.py

```diff
@@ -15,180 +13,178 @@
-        self.assertFalse(
-            bool(optional_present_deps),
-            f"Optional dependencies installed: {optional_present_deps}.\n"
-            "Please ensure that optional dependencies are not present in the core PyBaMM installation, "
-            "or list them as required.",
+    # Check that optional dependencies are not present in the core PyBaMM installation
+    optional_present_deps = optional_distribution_deps & required_distribution_deps
+    assert not bool(optional_present_deps), (
+        f"Optional dependencies installed: {optional_present_deps}.\n"
+        "Please ensure that optional dependencies are not present in the core PyBaMM installation, "
+        "or list them as required."
+    )
+
+
+def test_url_gets_to_stdout(mocker):
+    model = pybamm.BaseModel()
+    model.variables = {"Electrolyte concentration": 1, "Electrode potential": 0}
+
+    param = pybamm.ParameterValues({"test": 10, "b": 2})
+
+    # Test variables search (default returns key)
+    with mocker.patch("sys.stdout", new=StringIO()) as fake_out:
+        model.variables.search("Electrode")
+        assert fake_out.getvalue() == "Electrode potential\n"
+
+    # Test bad var search (returns best matches)
+    with mocker.patch("sys.stdout", new=StringIO()) as fake_out:
+        model.variables.search("Electrolyte cot")
+        out = (
+            "No results for search using 'Electrolyte cot'. "
+            "Best matches are ['Electrolyte concentration', "
+            "'Electrode potential']\n"
         )
+        assert fake_out.getvalue() == out
 
-
-class TestSearch(TestCase):
-    def test_url_gets_to_stdout(self):
-        model = pybamm.BaseModel()
-        model.variables = {"Electrolyte concentration": 1, "Electrode potential": 0}
-
-        param = pybamm.ParameterValues({"test": 10, "b": 2})
-
-        # Test variables search (default returns key)
-        with patch("sys.stdout", new=StringIO()) as fake_out:
-            model.variables.search("Electrode")
-            self.assertEqual(fake_out.getvalue(), "Electrode potential\n")
-
-        # Test bad var search (returns best matches)
-        with patch("sys.stdout", new=StringIO()) as fake_out:
-            model.variables.search("Electrolyte cot")
-            out = (
-                "No results for search using 'Electrolyte cot'. "
-                "Best matches are ['Electrolyte concentration', "
-                "'Electrode potential']\n"
-            )
-            self.assertEqual(fake_out.getvalue(), out)
-
-        # Test param search (default returns key, value)
-        with patch("sys.stdout", new=StringIO()) as fake_out:
-            param.search("test")
-            self.assertEqual(fake_out.getvalue(), "test\t10\n")
+    # Test param search (default returns key, value)
+    with mocker.patch("sys.stdout", new=StringIO()) as fake_out:
+        param.search("test")
+        assert fake_out.getvalue() == "test\t10\n"
 
 
 if __name__ == "__main__":
```

Comment:
Do we still need this block for being able to run the tests, or can `pytest` run individual files without it? Also, we run `pytest` with `-v` by default, so we probably don't need to check `sys.argv` either.

Reply:
Yes we can safely remove this block.